### PR TITLE
*: remove some unnecessary exported/interface functions

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: Related to #982.

Problem Summary: Remove some unnecessary functions to clean up the interface.

### What is changed and how it works?

- Remove `RebaseAutoID` from the `Table` interface.
- Remove `GenerateAutoID` and `GetAutoID`.

### Check List 

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

File: meta/meta_autoid.go [hunk 1]
```diff
@@ -132,8 +144,13 @@ func (a *autoIDAccessors) RowID() AutoIDAccessor {
 }
 
 // IncrementID is used to get the auto_increment ID meta key-value accessor.
-func (a *autoIDAccessors) IncrementID() AutoIDAccessor {
-	a.access.idEncodeFn = a.access.m.autoIncrementIDKey
+func (a *autoIDAccessors) IncrementID(tableVersion uint16) AutoIDAccessor {
+	// _tidb_rowid and auto_increment ID in old version TiDB share the same meta key-value.
+	if tableVersion < sepAutoIncVer {
```
Comment (zimulala) [inline meta/meta_autoid.go:149]: This version of `sepAutoIncVer` is not available yet?

Comment (tangenta) [reply meta/meta_autoid.go:149]: Yea, so the value is `model.TableInfoVersion4 + 1`. Later I will update it to `model.TableInfoVersion5`

**Review by AilinKid — APPROVED** [review]
Test?

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**